packages/apple-pay-integration/src/apple-pay-button-strategy.ts

```diff
@@ -204,12 +254,23 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
                 throw new Error('Merchant validation failed');
             }
         };
+        if (this._buyNowInitializeOptions && !this._requiresShipping) {
+            applePaySession.onpaymentmethodselected = async () => {
+                await this._createBuyNowCart();
+                return this._handlePaymentMethodSelected(applePaySession);
+            };
+        }
 
-        applePaySession.onshippingcontactselected = async (event) =>
-            this._handleShippingContactSelected(applePaySession, config, event);
+        applePaySession.onshippingcontactselected = async (event) => {
+            if (this._buyNowInitializeOptions && this._requiresShipping) {
+                await this._createBuyNowCart();
+            }
+            return this._handleShippingContactSelected(applePaySession, storeName, event);
+        };
 
-        applePaySession.onshippingmethodselected = async (event) =>
-            this._handleShippingMethodSelected(applePaySession, config, event);
+        applePaySession.onshippingmethodselected = async (event) => {
+            return this._handleShippingMethodSelected(applePaySession, storeName, event);
```

Comment:
Do we need to return anything here? As far as I see, there is no return in prev implementation.

Reply:
fixed